packages/typespec-client-generator-core/src/types.ts

```diff
@@ -114,30 +119,20 @@ function addFormatInfo(
   type: ModelProperty | Scalar,
   propertyType: SdkType
 ): void {
-  const format = getFormat(context.program, type)?.toLocaleLowerCase();
-  if (format) {
-    switch (format) {
-      case "guid":
-      case "uuid":
-      case "password":
-      case "etag":
-        propertyType.kind = format;
-        break;
-      case "url":
-      case "uri":
-        propertyType.kind = "url";
-        break;
-      case "armid":
-        propertyType.kind = "armId";
-        break;
-      case "ipaddress":
-        propertyType.kind = "ipAddress";
-        break;
-      case "azurelocation":
-        propertyType.kind = "azureLocation";
-        break;
-      default:
-        break;
+  const format = getFormat(context.program, type);
```

Comment:
Can we keep the map on the left exactly as it is with a `legacyFormat` function. Otherwise here as we keep adding new known scalar `@format` with the same name will work even though it shouldn't